src/main/kotlin/it/pagopa/wallet/client/NpgClient.kt

```diff
@@ -36,74 +53,128 @@ class NpgClient(
-            logger.error(
-                "Error communicating with NPG-orderBuild for correlationId $correlationId - response: ${it.responseBodyAsString}",
-                it
-            )
-            mapNpgException(it.statusCode)
+        return usingNpgTracing(
+            tracer,
+            GatewayOperation.BUILD_FORM,
+            { it.setAttribute(NPG_CORRELATION_ID_ATTRIBUTE_NAME, correlationId.toString()) }
+        ) { span, operation ->
+            logger.info("Sending orderBuild with correlationId: $correlationId")
+            Try.of {
+                    apiKey.fold(
+                        { Mono.error(it) },
+                        {
+                            npgWebClient.pspApiV1OrdersBuildPost(
+                                correlationId,
+                                it,
+                                createHostedOrderRequest
+                            )
+                        }
+                    )
+                }
+                .getOrElseGet { Mono.error(it) }
+                .doOnError(WebClientResponseException::class.java) {
+                    logger.error(
+                        "Error communicating with NPG-orderBuild  for correlationId $correlationId - response: ${it.responseBodyAsString}",
+                        it
+                    )
+                }
+                .onErrorMap { error -> exceptionToNpgResponseException(error, span, operation) }
         }
     }
 
     fun getCardData(sessionId: String, correlationId: UUID): Mono<CardDataResponse> {
-        val response: Mono<CardDataResponse> =
-            try {
-                logger.info("getCardData with correlationId: $correlationId")
-                npgWebClient.pspApiV1BuildCardDataGet(
-                    correlationId,
-                    npgPaypalPspApiKeysConfig.defaultApiKey,
-                    sessionId
-                )
-            } catch (e: WebClientResponseException) {
-                Mono.error(e)
-            }
-        return response.onErrorMap(WebClientResponseException::class.java) {
-            logger.error(
-                "Error communicating with NPG-getCardData for correlationId $correlationId - response: ${it.responseBodyAsString}",
-                it
-            )
-            mapNpgException(it.statusCode)
+        return usingNpgTracing(
+            tracer,
+            GatewayOperation.GET_CARD_DATA,
+            { it.setAttribute(NPG_CORRELATION_ID_ATTRIBUTE_NAME, correlationId.toString()) }
+        ) { span, operation ->
+            logger.info("getCardData with correlationId: $correlationId")
+            Try.of {
+                    npgWebClient.pspApiV1BuildCardDataGet(
+                        correlationId,
+                        npgPaypalPspApiKeysConfig.defaultApiKey,
+                        sessionId
+                    )
+                }
+                .getOrElseGet { Mono.error(it) }
```

Comment:
Is `Try` necessary? Exceptions should be already caught in the reactive chain

Reply:
In other words, these should be equivalent, right?
```kotlin
fun foo(): Mono<Integer> {
    Mono.just(0).map {
        throw RuntimeException()
    }
}

val m = Try.of { foo() }.getOrElseGet { Mono.error(it) }
val m2 = foo()
```

Reply:
Solved by https://github.com/pagopa/pagopa-wallet-service/pull/167/commits/3a598cdb763f8f2a6bfe1ecebaede63c01311e9d